### PR TITLE
RFR PAusers

### PR DESCRIPTION
**Pull request by SensitiveLion**
Added users profile page and index page

### CODE REVIEW

File: spec/features/user_profiles_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+require 'helpers'
+
+feature 'user can view profile' do
+  let(:game) { FactoryGirl.create(:game) }
+  let(:user) { FactoryGirl.create(:user) }
+
+  scenario 'users can view their profile' do
+    sign_in_as(user)
+    make_review_for_game(game)
```
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]: What is the benefit of this helper versus just doing something like 

```ruby
review = FactoryGirl.create(:review)
```

Also do you need to create a review to see a user profile? It shouldn't be the case that that is necessary.

File: spec/features/user_profiles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    visit users_path
+    within(".column") do
+      click_link user.username
+    end
+    expect(page).to have_content(user.username)
+    expect(page).to have_content("review of #{game.name}")
+  end
```
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]: This test can just be pseudocoded like this:

```
visit user path of a user
expect to see a bunch of that user's data
```

File: spec/features/user_profiles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    sign_in_as(user)
+    make_review_for_game(game)
+    click_link "Sign Out"
+    visit users_path
+    within(".column") do
+      click_link user.username
+    end
+    expect(page).to have_content(user.username)
+    expect(page).to have_content("review of #{game.name}")
+  end
```
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]: Same comment as above.

File: spec/features/user_profiles_spec.rb (hunk 1 above)
Comment (psgoodman) [inline spec/features/user_profiles_spec.rb]: That helper lets us be sure that the review is created by the user we're using. And having a review ensures that reviews are showing up on the profile the way they should. We could make that a separate test if you think we should?

File: spec/features/user_profiles_spec.rb (hunk 2 above)
Comment (psgoodman) [inline spec/features/user_profiles_spec.rb]: I'm not sure I follow. Are you saying we don't need to actually write out the test? Or just that we could simplify it? Or something else?

File: spec/features/user_profiles_spec.rb (hunk 1 above)
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]:
```ruby
user = FactoryGirl.create(:user)
review = FactoryGirl.create(:review, user: user)
```

File: spec/features/user_profiles_spec.rb (hunk 2 above)
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]: Simplify. Testing shows expectations for specific scenarios. If you're scenario is a user can visit their own profile, then the test would be this:

```ruby
sign_in_as(user)
visit user_path(user)
expect(page).to have_content(user.username)
```

You don't need to check anything review related unless your scenario had to do with that (like 'see review of user' or something).

File: spec/features/user_profiles_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      expect(page).to have_content(user.username)
+    end
+  end
+end
+
+feature 'user can view profile' do
+  let(:review) { FactoryGirl.create(:review, user: user) }
+  let(:user) { FactoryGirl.create(:user) }
+
+  scenario 'users can view their profile' do
```
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]: If you want to write the test below name this test as `scenario 'user can navigate to their own profile from the users index page'`.

File: spec/features/user_profiles_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    within(".column") do
+      click_link user.username
+    end
+    expect(page).to have_content(user.first_name)
+    expect(page).to have_content(user.last_name)
+    expect(page).to have_content(user.age)
+    expect(page).to have_content(user.location)
+  end
+
+  scenario 'visitors can view profiles' do
```
Comment (vikram7) [inline spec/features/user_profiles_spec.rb]: `scenario 'visitor can navigate to profiles from the users index page'`

File: coverage/.last_run.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 76.31
+    "covered_percent": 77.57
   }
 }
```
Comment (atsheehan) [inline coverage/.last_run.json]: The `/coverage` directory should not be included in your repository. You can remove it with:

```no-highlight
$ git rm -rf coverage
```

And then add it to your `.gitignore` file:

```no-highlight
/coverage
```

**Pull request merged**